src/Workspaces/CSharp/Portable/CodeGeneration/PropertyGenerator.cs

```diff
@@ -367,7 +367,7 @@ private static BlockSyntax GenerateBlock(IMethodSymbol accessor)
                     // because of the boxing requirement in order to call the method.
                     // therefore it seems like a small oversight to leave out the keyword for an explicit impl from metadata.
                     var hasAllReadOnlyAccessors = property.GetMethod?.IsReadOnly != false && property.SetMethod?.IsReadOnly != false;
-                    if (hasAllReadOnlyAccessors)
+                    if (hasAllReadOnlyAccessors && !property.ContainingType.IsReadOnly)
```

Comment:
1. you did property.ContainingTYpe here, but not method.ContainingType above.  Why?
2. we shoudl doc the intuition here.